core/src/main/java/org/sonarsource/sonarlint/core/log/MessageFormat.java

```diff
@@ -0,0 +1,491 @@
+    } else {
+      return false;
+    }
+    return true;
+  }
+
+  private static boolean appendDate(final Object o, final StringBuilder str) {
+    if (!(o instanceof Date)) {
+      return false;
+    }
+    final Date date = (Date) o;
+    final SimpleDateFormat format = getSimpleDateFormat();
+    str.append(format.format(date));
+    return true;
+  }
+
+  private static SimpleDateFormat getSimpleDateFormat() {
+    SimpleDateFormat result = threadLocalSimpleDateFormat.get();
+    if (result == null) {
+      result = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+      threadLocalSimpleDateFormat.set(result);
+    }
+    return result;
+  }
+
+  /**
+   * Returns {@code true} if the specified object is an array, a Map or a Collection.
+   */
+  private static boolean isMaybeRecursive(final Object o) {
+    return o.getClass().isArray() || o instanceof Map || o instanceof Collection;
+  }
+
+  private static void appendPotentiallyRecursiveValue(final Object o, final StringBuilder str,
+    final Set<String> dejaVu) {
+    final Class<?> oClass = o.getClass();
+    if (oClass.isArray()) {
+      appendArray(o, str, dejaVu, oClass);
+    } else if (o instanceof Map) {
+      appendMap(o, str, dejaVu);
+    } else if (o instanceof Collection) {
+      appendCollection(o, str, dejaVu);
+    }
+  }
+
+  private static void appendArray(final Object o, final StringBuilder str, @Nullable Set<String> dejaVu, final Class<?> oClass) {
+    if (oClass == byte[].class) {
+      str.append(Arrays.toString((byte[]) o));
+    } else if (oClass == short[].class) {
+      str.append(Arrays.toString((short[]) o));
+    } else if (oClass == int[].class) {
+      str.append(Arrays.toString((int[]) o));
+    } else if (oClass == long[].class) {
+      str.append(Arrays.toString((long[]) o));
+    } else if (oClass == float[].class) {
+      str.append(Arrays.toString((float[]) o));
+    } else if (oClass == double[].class) {
+      str.append(Arrays.toString((double[]) o));
+    } else if (oClass == boolean[].class) {
+      str.append(Arrays.toString((boolean[]) o));
+    } else if (oClass == char[].class) {
+      str.append(Arrays.toString((char[]) o));
+    } else {
+      if (dejaVu == null) {
+        dejaVu = new HashSet<>();
+      }
+      // special handling of container Object[]
+      final String id = identityToString(o);
+      if (dejaVu.contains(id)) {
+        str.append(RECURSION_PREFIX).append(id).append(RECURSION_SUFFIX);
+      } else {
+        dejaVu.add(id);
+        final Object[] oArray = (Object[]) o;
+        str.append('[');
+        boolean first = true;
+        for (final Object current : oArray) {
+          if (first) {
+            first = false;
+          } else {
+            str.append(", ");
+          }
+          recursiveDeepToString(current, str, new HashSet<>(dejaVu));
+        }
+        str.append(']');
+      }
+      // str.append(Arrays.deepToString((Object[]) o));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)